### ISSUE TITLE
KeyError exceptions when a sensor disappears temporarily

### ISSUE THREAD

**Issue by knyar** [MindrustUK/Heatmiser-for-home-assistant#183]
I noticed recently that my script in Home Assistant could not set target temperature for the thermostats.
Looking at the logs, I saw an exception getting raised every time Heatmiser entities are updated:

```
2024-10-13 10:14:14.005 INFO (MainThread) [custom_components.heatmiserneo] Executing update_data()
2024-10-13 10:14:14.565 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 258, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 453, in _async_refresh
    self.async_update_listeners()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 168, in async_update_listeners
    update_callback()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 533, in _handle_coordinator_update
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1005, in async_write_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1130, in _async_write_ha_state
    self.__async_calculate_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1067, in __async_calculate_state
    state = self._stringify_state(available)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1011, in _stringify_state
    if (state := self.state) is None:
                 ^^^^^^^^^^
  File "/config/custom_components/heatmiserneo/switch.py", line 81, in state
    return 'on' if self.data.timer_on else 'off'
                   ^^^^^^^^^
  File "/config/custom_components/heatmiserneo/switch.py", line 61, in data
    return timers[self.name]
           ~~~~~~^^^^^^^^^^^
KeyError: 'Living Room Sensor'
```

The sensor in question is a battery-powered temperature sensor. I suspect it might have disconnected temporarily, and Neohub stopped returning data for it.

Looking at the code of the integration, it seems like it expects thermostats to never be renamed or removed. Is this intentional? I think it might be pretty easy to keep returning previous state and mark the entity as unavailable if we no longer see it in the response from Neohub.

**Comment by MindrustUK** [comment]
Thanks for the pull request. I'll review in due course. With regards to thermostats never being renamed or removed, it's not intentional, it's another one of the many quality of life features I've ignored for now in favour of addressing other items. In practical terms I'm not sure how much people in daily use would be removing devices regularly so I've not put any serious focus on this.

**Comment by MindrustUK** [comment]
Hi @knyar I've had a quick review of the change and this is against the now deprecated version 1 branch. I'll take your changes and roll them into to Dev for testing and and version 1.5 once they've been been proven.

**Comment by ocrease** [comment]
@knyar there have been a lot of changes on the branch now in HACS and a lot more changes on the dev branch. It would be interesting to know if you still get these errors, specially on the dev branch.

**Comment by ocrease** [comment]
@MindrustUK I think this can be closed now that #216 has been merged into dev
